snippets/javascript/framer/framer.json

```diff
@@ -0,0 +1,176 @@
+      "    initial=\"hidden\"",
+      "    animate=\"visible\"",
+      "    variants={${2:variants}}",
+      "  ></motion.${1:div}>"
+    ],
+    "scope": "javascript,javascriptreact",
+    "description": "Motion shares the simple animation and gesture API as the Framer X library, while maintaining HTML and SVG semantics. This makes it simple to port your high-fidelity prototypes to a production-ready environment, or make something completely new!"
+  },
+  "framer-variants": {
+    "prefix": ["framer variants", "variants"],
+    "body": [
+      "  const ${1:variants} = {",
+      "    hidden: {",
+      "      opacity: 0,",
+      "    },",
+      "    visible: {",
+      "      opacity: 1,",
+      "      transition: {",
+      "        delay: ${2:1},",
+      "        duration: ${3:1},",
+      "      },",
+      "    },",
+      "  };"
+    ],
+    "scope": "javascript,javascriptreact",
+    "description": "They're passed into motion components via the variants prop. <motion.div variants={variants} /> These variants can be referred to by label, wherever you can set an animation target."
+  },
+  "framer-sequence": {
+    "prefix": ["framer sequence"],
+    "body": [
+      "const ${1:sequence} = async () => {",
+      "  await ${2:animateControls}.start({ ${3:x: 0} })",
+      "  return await ${4:itemControls}.start({ ${5:opacity: 1} })",
+      "}"
+    ],
+    "scope": "javascript,javascriptreact",
+    "description": "start returns a Promise, so it can be used to sequence animations using await or then. Different controls can be sequenced together, and these sequences can be composed into functions that themselves can then be sequenced."
+  },
+  "framer-use-animation": {
+    "prefix": ["framer use-animation"],
+    "body": ["const ${controls} = useAnimation()"],
+    "scope": "javascript,javascriptreact",
+    "description": "useMotionValue(initial) Declarative animations are ideal for most UI interactions. But sometimes we need to orchestrate more complex sequences. The useAnimation hook can be used to create a set of imperative AnimationControls with a start and stop method. These controls can be passed to one or more motion components via the animate prop."
+  },
+  "framer-use-motion-value": {
+    "prefix": ["framer use-motion-value"],
+    "body": ["const ${1:scale} = useMotionValue(${2:1})"],
+    "scope": "javascript,javascriptreact",
+    "description": "Creates a MotionValue to track the state and velocity of a value. Usually, these are created automatically. For advanced use-cases, like use with useTransform, you can create MotionValues externally and pass them into the animated component via the style prop."
+  },
+  "framer-use-transform": {
+    "prefix": ["framer use-transform"],
+    "body": [
+      "//const x = useMotionValue(0)",
+      "//const xRange = [-200, -100, 100, 200]",
+      "//const opacityRange = [0, 1, 1, 0]",
+      "const ${1:opacity} = useTransform(${2:x}, ${3:xRange}, ${4:opacityRange})"
```

Comment:
wrong example based on the documentation you have one option with range and another one with the function.